src/extension.ts

```diff
@@ -86,9 +86,15 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 
     restartInProgress = true;
 
-    if (nativeServer) {
-      traceVerbose("Using experimental server with bundled Ruff binary");
-
+    const interpreter = getInterpreterFromSetting(serverId);
+    if (
+      interpreter &&
+      interpreter.length > 0 &&
```

Comment:
```suggestion
      interpreter?.length > 0 &&
```

Reply:
or does typescript complain that the left isn't a number. In that case I would write `interpreter != null && interpreter.length > 0`. We should probably enable the lint rule to enforce `!=` instead of on truthiness checks.

Reply:
Yeah, the compiler complains. I'll change it to use explicit `null` checks but I can't find the lint rule which corresponds to that in https://www.typescriptlang.org/tsconfig/.

Reply:
I think it's https://typescript-eslint.io/rules/strict-boolean-expressions/